src/test/java/service/impl/FruitShopServiceImplTest.java

```diff
@@ -0,0 +1,89 @@
+package service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import db.Storage;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import service.FruitShopService;
+import strategy.OperationStrategy;
+import strategy.OperationStrategyImpl;
+import strategy.TransactionHandler;
+import strategy.handler.BalanceHandler;
+import strategy.handler.PurchaseHandler;
+import strategy.handler.ReturnHandler;
+import strategy.handler.SupplyHandler;
+
+public class FruitShopServiceImplTest {
+    private static final String FRUIT_FOR_TEST = "banana";
+    private static OperationStrategy operationStrategy;
+    private static FruitShopService fruitShopService;
+    private static Map<FruitTransaction.Operation, TransactionHandler> handlerMap;
+    private static TransactionHandler balanceHandler;
+    private static TransactionHandler returnHandler;
+    private static TransactionHandler purchaseHandler;
+    private static TransactionHandler suppluHandler;
```

Comment:
You really don't need here so many dependencies, cause its unit testing. It's enough to have only one operation here (For example Balance). The main purpose and goal are to test the functionality of FruitShopServiceImpl class. No related classes